tensorflow/python/kernel_tests/cwise_ops_test.py

```diff
@@ -55,6 +56,30 @@
 _XOR = lambda x, y: x ^ y
 _INV = lambda x: ~x
 
+# @ operator supported since python 3.5.
+if sys.version_info >= (3, 5):
+  # Use eval() to avoid a compilation error on earlier versions.
+  _MATMUL = lambda x, y: eval("x @ y")
+else:
+  # For earlier versions of python, emulate regular behavior.
+  # Useful to build and test for 3.5+ on earlier versions.
+  def _MATMUL(x, y):
```

Comment:
I did my best to emulate the @ operator properly so the tests would work whether on python 3.5+ or with earlier versions.  Simply calling `type(x).__matmul__(x, y)` obviously does not work when `x` is a python array (since it is not defined), but it does not work either when `x` is a NumPy array, because it returns `NotImplemented` when `y` is not a NumPy array and it implements `__rmatmul__()`.  Lines 74 to 80 are unused by the current tests, but this code is simple enough and allows adding more tests without bad surprises, so I think it shouldn't be shortened.